modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js

```diff
@@ -144,15 +144,38 @@ export default function DatePicker({
 
 	const [expanded, setExpanded] = useState(false);
 
+	const handleBlur = () => {
+		onBlur?.();
+		if (!otherProps.required) {
+			const isValidMomentFormat = moment(
+				formattedDate,
+				momentFormat,
+				true
+			).isValid();
+
+			const isFill = /\d/.test(formattedDate);
+
+			if (!isFill || isValidMomentFormat) {
+				setMomentFormatValid(false);
+
+				return;
+			}
+			if (!isValidMomentFormat || (!expanded && !isValidMomentFormat)) {
+				setMomentFormatValid(true);
+			}
+		}
+	};
```

Comment:
Hey @vngoliveira, you can see more details about the moment parameters here:

1. https://momentjs.com/docs/#/parsing/string-format/
2. https://momentjs.com/guides/#/parsing/strict-mode/

To understand more about these parameters. A blank value should not be validated as true when compared to the mask we are using. So we can remove the isFill validation, and improve the `if` logic. I suggest we reorganize this function, I had the opportunity to make a sketch. Could you please carry out the necessary tests:

```
	const handleBlur = () => {
		onBlur?.();

		if (!otherProps.required) {
			const isValidMomentFormat = moment(
				formattedDate,
				momentFormat,
				true
			).isValid();

			if(isValidMomentFormat) {
				setMomentFormatInvalid(false)

				return 
			}
			
			setMomentFormatInvalid(true);
		}
	};
```